tap_google_ads/streams.py

```diff
@@ -56,6 +64,52 @@ def get_request_timeout(config):
         request_timeout = DEFAULT_REQUEST_TIMEOUT
     return request_timeout
 
+
+def get_split_by_period(config) -> relativedelta | None:
+    """Constructs the date range for a single API request from config and errors
+    on invalid values.
+
+    The value has the following schema:
+
+    ```
+    "split_by_period": {
+      "days": <int>,
+      "month": <int>,
+      "years": <int>
+    } 
+    ```
+
+    At least "days" or "months" or "years" must be set and be greater than 0.
+
+    If value is set, it will be used to split the requested period into series
+    of date ranges and each date range will be requested separately.
+
+    If value is not set or can't be correcly parsed, the function returns `None`
+    and the tap fetches the whole requested period in one request.
+    """
+
+    split_by_period = config.get("split_by_period")
+
+    if split_by_period is None:
+        return None
+
+    days = split_by_period.get("days") or "0"
+    months = split_by_period.get("months") or "0"
+    years = split_by_period.get("years") or "0"
+
+    try:
+        period = relativedelta(
+            days = int(days),
+            months = int(months),
+            years = int(years),
+        )
+    except (ValueError, TypeError):
+        LOGGER.warning(f"Provided split_by_period value is invalid it will be set to None.")
```

Comment:
this f-string does not use any interpolated variables. Let's include erroneous split_by_period in the message.

Reply:
Fixed. Thanks.